packages/buidler-core/src/builtin-tasks/compile.ts

```diff
@@ -390,380 +508,435 @@ export default function () {
+      }: {
+        compilationJob: CompilationJob;
+        emittedArtifactsPerFile: EmittedArtifactsPerFile;
+        quiet: boolean;
+      }) => {
+        if (quiet) {
+          return;
+        }
+
+        const numberOfContracts = emittedArtifactsPerFile.length;
+        console.log(
+          "Compiled",
+          numberOfContracts,
+          pluralize(numberOfContracts, "contract"),
+          "successfully"
+        );
+      }
+    );
 
   /**
    * This is an orchestrator task that uses other internal tasks to compile a
-   * compilation group.
+   * compilation job.
    */
-  internalTask(
-    TASK_COMPILE_COMPILE_GROUP,
-    async (
-      {
-        compilationGroup,
-        solidityFilesCache,
-        force,
-      }: {
-        compilationGroup: CompilationGroup;
-        solidityFilesCache?: SolidityFilesCache;
-        force: boolean;
-      },
-      { run }
-    ) => {
-      log(`Compiling group with version '${compilationGroup.getVersion()}'`);
-      await run(TASK_COMPILE_LOG_COMPILE_GROUP_START, { compilationGroup });
-
-      const input: SolcInput = await run(TASK_COMPILE_GET_COMPILER_INPUT, {
-        compilationGroup,
-      });
-
-      const output = await run(TASK_COMPILE_COMPILE, {
-        solcVersion: compilationGroup.solidityConfig.version,
-        input,
-      });
+  internalTask(TASK_COMPILE_SOLIDITY_COMPILE_JOB)
+    .addParam("compilationJob", undefined, undefined, types.any)
+    .addParam("quiet", undefined, undefined, types.boolean)
+    .setAction(
+      async (
+        {
+          compilationJob,
+          quiet,
+        }: {
+          compilationJob: CompilationJob;
+          quiet: boolean;
+        },
+        { run }
+      ): Promise<{ emittedArtifactsPerFile: EmittedArtifactsPerFile }> => {
+        log(
+          `Compiling job with version '${
+            compilationJob.getSolcConfig().version
+          }'`
+        );
+        await run(TASK_COMPILE_SOLIDITY_LOG_COMPILE_JOB_START, {
+          compilationJob,
+          quiet,
+        });
 
-      await run(TASK_COMPILE_CHECK_ERRORS, { output });
+        const input: SolcInput = await run(
+          TASK_COMPILE_SOLIDITY_GET_COMPILER_INPUT,
+          {
+            compilationJob,
+          }
+        );
 
-      if (output === undefined) {
-        log(`No output for compilation group`);
-        return;
-      }
+        const output = await run(TASK_COMPILE_SOLIDITY_COMPILE, {
+          solcVersion: compilationJob.getSolcConfig().version,
+          input,
+        });
 
-      const { numberOfContracts } = await run(TASK_COMPILE_EMIT_ARTIFACTS, {
-        compilationGroup,
-        input,
-        output,
-        solidityFilesCache,
-        force,
-      });
+        await run(TASK_COMPILE_SOLIDITY_CHECK_ERRORS, { output, quiet });
 
-      await run(TASK_COMPILE_LOG_COMPILE_GROUP_END, {
-        compilationGroup,
-        numberOfContracts,
-      });
-    }
-  );
+        if (output === undefined) {
```

Comment:
I just checked, and this can't be true. `output` is the output of `Compiler#compile`.